spec/server/functional/user.spec.coffee

```diff
@@ -218,17 +218,17 @@ describe 'PUT /db/user', ->
     [res, body] = yield request.putAsync { uri: url, json: { role: 'student' }}
     expect(body.role).toBe('advisor')
 
-  it 'returns 422 if both email and name would be unset for a registered user', utils.wrap ->
-    user = yield utils.initUser()
+  it 'returns 422 if both email and name would be unset for a student', utils.wrap ->
+    user = yield utils.initUser({ role: 'student' })
     yield utils.loginUser(user)
     [res, body] = yield request.putAsync { uri: getURL('/db/user/'+user.id), json: { email: '', name: '' }}
     expect(body.code).toBe(422)
     expect(body.message).toEqual('User needs a username or email address')
     
-  it 'allows unsetting email, even when there\'s a user with email and emailLower set to empty string', utils.wrap ->
+  it 'allows unsetting email on student accounts, even when there\'s a user with email and emailLower set to empty string', utils.wrap ->
     invalidUser = yield utils.initUser()
     yield invalidUser.update({$set: {email: '', emailLower: ''}})
-    user = yield utils.initUser()
+    user = yield utils.initUser({role: 'student'})
     yield utils.loginUser(user)
     [res, body] = yield request.putAsync { uri: getURL('/db/user/'+user.id), json: { email: '' }}
     expect(res.statusCode).toBe(200)
```

Comment:
Should add a test here for making sure when you have an email, then remove it as an individual, you get a 422.